Bottom: 77d617c666a85a480e2df949197e53c8a0c3e15d
Top:    91a0543f5f175abae8158d37e8612790498fd113
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-04-10 15:15:02 -0500

Use docker named volumes

Instead of a data container, use docker named volume


---
diff --git a/roles/docker-exim4/tasks/main.yml b/roles/docker-exim4/tasks/main.yml
index f996918..6236d76 100644
--- a/roles/docker-exim4/tasks/main.yml
+++ b/roles/docker-exim4/tasks/main.yml
@@ -92,10 +92,8 @@
 # Create the data volumes
 
 - name: data-volume container
-  docker_datavolume:
-    image_name: '{{ exim4_image_name }}'
-    image_tag: '{{ docker_image_tag }}'
-    data_volume_container_name: '{{ exim4_dv_name }}'
+  docker_volume:
+    name: '{{ exim4_dv_name }}'
 
 # *****************************************************************************
 # Start the data container running
diff --git a/roles/docker-gitlab/tasks/main.yml b/roles/docker-gitlab/tasks/main.yml
index 0da097e..0e7407b 100644
--- a/roles/docker-gitlab/tasks/main.yml
+++ b/roles/docker-gitlab/tasks/main.yml
@@ -97,16 +97,12 @@
 # Create the data volumes
 
 - name: data-volume container (gitlab)
-  docker_datavolume:
-    image_name: '{{ gitlab_dv_image_name }}'
-    image_tag: '{{ docker_image_tag }}'
-    data_volume_container_name: '{{ gitlab_dv_name }}'
+  docker_volume:
+    name: '{{ gitlab_dv_name }}'
 
 - name: data-volume container (postgres)
-  docker_datavolume:
-    image_name: '{{ gitlab_db_image_name }}'
-    image_tag: '{{ docker_image_tag }}'
-    data_volume_container_name: '{{ gitlab_db_dv_name }}'
+  docker_volume:
+    name: '{{ gitlab_db_dv_name }}'
 
 - name: stop prev container (postgres)
   docker_container:
@@ -140,7 +136,7 @@
     name: '{{ gitlab_db_container_name }}'
     detach: true
     restart_policy: always
-    volumes-from: '{{ gitlab_db_dv_name }}'
+    volumes_from: '{{ gitlab_db_dv_name }}'
     image: '{{ gitlab_db_image_name }}:{{ docker_image_tag }}'
 
 - name: start container (gitlab)
diff --git a/roles/docker-mantisbt/tasks/main.yml b/roles/docker-mantisbt/tasks/main.yml
index 2124a54..3352d62 100644
--- a/roles/docker-mantisbt/tasks/main.yml
+++ b/roles/docker-mantisbt/tasks/main.yml
@@ -110,16 +110,12 @@
 # Create the data volumes
 
 - name: data-volume container (mantisbt)
-  docker_datavolume:
-    image_name: '{{ mantisbt_image_name }}'
-    image_tag: '{{ docker_image_tag }}'
-    data_volume_container_name: '{{ mantisbt_dv_name }}'
+  docker_volume:
+    name: '{{ mantisbt_dv_name }}'
 
 - name: data-volume container (mysql)
-  docker_datavolume:
-    image_name: '{{ mantisbt_db_image_name }}'
-    image_tag: '{{ docker_image_tag }}'
-    data_volume_container_name: '{{ mantisbt_db_dv_name }}'
+  docker_volume:
+    name: '{{ mantisbt_db_dv_name }}'
 
 - name: stop prev container (mysql)
   docker_container:
@@ -149,7 +145,7 @@
     name: '{{ mantisbt_db_container_name }}'
     detach: true
     restart_policy: always
-    volumes-from:
+    volumes_from:
       - '{{ mantisbt_db_dv_name }}'
     image: '{{ mantisbt_db_image_name }}:{{ docker_image_tag }}'
 
diff --git a/roles/docker-mediawiki/tasks/main.yml b/roles/docker-mediawiki/tasks/main.yml
index cfcd636..8746377 100644
--- a/roles/docker-mediawiki/tasks/main.yml
+++ b/roles/docker-mediawiki/tasks/main.yml
@@ -87,16 +87,12 @@
 # Create the data volumes
 
 - name: data-volume container (mediwiki)
-  docker_datavolume:
-    image_name: '{{ wiki_image_name }}'
-    image_tag: '{{ docker_image_tag }}'
-    data_volume_container_name: '{{ wiki_dv_name }}'
+  docker_volume:
+    name: '{{ wiki_dv_name }}'
 
 - name: data-volume container (mysql)
-  docker_datavolume:
-    image_name: '{{ wiki_db_image_name }}'
-    image_tag: '{{ docker_image_tag }}'
-    data_volume_container_name: '{{ wiki_db_dv_name }}'
+  docker_volume:
+    name: '{{ wiki_db_dv_name }}'
 
 - name: stop prev container (mysql)
   docker_container:
@@ -123,7 +119,7 @@
     name: '{{ wiki_db_container_name }}'
     detach: true
     restart_policy: always
-    volumes-from: '{{ wiki_db_dv_name }}'
+    volumes_from: '{{ wiki_db_dv_name }}'
     image: '{{ wiki_db_image_name }}:{{ docker_image_tag }}'
 
 - name: start container (mediwiki)
@@ -140,11 +136,11 @@
     volumes-from: 
       - '{{ openssl_dv_name }}'
       - '{{ wiki_dv_name }}'
-      env:
-        - WIKI_HOSTNAME={{ container_addr_map.wiki.hostname }}
-        - WIKI_MAIL_USER=wiki
-        - WIKI_MAIL_PASSWORD={{ wiki_email_password | quote }}
-      image: '{{ wiki_image_name }}:{{ docker_image_tag }}'
+    env:
+      WIKI_HOSTNAME: '{{ container_addr_map.wiki.hostname }}'
+      WIKI_MAIL_USER: 'wiki'
+      WIKI_MAIL_PASSWORD: '{{ wiki_email_password | quote }}'
+    image: '{{ wiki_image_name }}:{{ docker_image_tag }}'
 
 # *****************************************************************************
 # restore?
diff --git a/roles/docker-openldap/tasks/main.yml b/roles/docker-openldap/tasks/main.yml
index c84183a..2d0c723 100644
--- a/roles/docker-openldap/tasks/main.yml
+++ b/roles/docker-openldap/tasks/main.yml
@@ -76,11 +76,13 @@
 # *****************************************************************************
 # Create the data volumes
 
-- name: data-volume container
-  docker_datavolume:
-    image_name: '{{ openldap_image_name }}'
-    image_tag: '{{ docker_image_tag }}'
-    data_volume_container_name: '{{ openldap_dv_name }}'
+- name: config volume
+  docker_volume:
+    name: '{{ openldap_cv_name }}'
+
+- name: data volume
+  docker_volume:
+    name: '{{ openldap_dv_name }}'
 
 # *****************************************************************************
 # Populate the ldap data-volume with default data
@@ -92,7 +94,8 @@
     image: '{{ openldap_image_name }}:{{ docker_image_tag }}'
     command: ['init_data_volumes']
     volumes:
-      - '{{ openldap_dv_name }}'
+      - '{{ openldap_cv_name }}:/etc/ldap:z'
+      - '{{ openldap_dv_name }}:/var/lib/ldap:z'
   when: st_openldap_restore.stat.exists == False
 
 - name: initial populate
@@ -110,9 +113,10 @@
     restart_policy: always
     hostname: '{{ container_addr_map.ldap.hostname }}'
     ports: '{{ container_port_map.ldap.port_args }}'
-    volumes-from:
-      - '{{ openssl_dv_name }}'
-      - '{{ openldap_dv_name }}'
+    volumes:
+      - '{{ openssl_dv_name }}:/etc/ssl/private:z'
+      - '{{ openldap_cv_name }}:/etc/ldap:z'
+      - '{{ openldap_dv_name }}:/var/lib/ldap:z'
     env:
       LDAP_HOSTNAME: '{{ container_addr_map.ldap.hostname }}'
     image: '{{ openldap_image_name }}:{{ docker_image_tag }}'
@@ -124,8 +128,8 @@
     restart_policy: always
     hostname: '{{ container_addr_map.ldap.hostname }}'
     ports: '{{ container_port_map.phpldapadmin.port_args }}'
-    volumes-from:
-      - '{{ openssl_dv_name }}'
+    volumes:
+      - '{{ openssl_dv_name }}:/etc/ssl/private:z'
     env:
       PHPLDAPADMIN_HTTPS: 'false'
       PHPLDAPADMIN_LDAP_HOSTS: 'ldap'
diff --git a/roles/docker-openldap/tasks/restore.yml b/roles/docker-openldap/tasks/restore.yml
index 27c0536..8e4a335 100644
--- a/roles/docker-openldap/tasks/restore.yml
+++ b/roles/docker-openldap/tasks/restore.yml
@@ -34,7 +34,7 @@
   command: 'docker stop {{ openldap_container_name }}'
 
 - name: load ldif files
-  command: 'docker run --rm --volumes-from {{ openldap_dv_name }} -v {{ host_openldap_docker_restore_dir }}:/tmp/import_export {{ openldap_image_name }}:{{ docker_image_tag }} apply_ldif -n {{ item.dbnum }} -l {{ item.ldif_file }}'
+  command: 'docker run --rm --volume {{ openldap_cv_name }}:/etc/ldap:z --volume {{ openldap_dv_name }}:/var/lib/ldap:z -v {{ host_openldap_docker_restore_dir }}:/tmp/import_export {{ openldap_image_name }}:{{ docker_image_tag }} apply_ldif -n {{ item.dbnum }} -l {{ item.ldif_file }}'
   with_items: '{{ openldap_backup_files }}'
 
 - name: restart container (ldap)
diff --git a/roles/docker-openldap/vars/main.yml b/roles/docker-openldap/vars/main.yml
index 3dd1df3..8e02ed0 100644
--- a/roles/docker-openldap/vars/main.yml
+++ b/roles/docker-openldap/vars/main.yml
@@ -8,6 +8,7 @@ openldap_image_name: '{{ docker_registry_username }}/ldap'
 openldap_container_name: '{{ docker_name_prefix }}ldap'
 
 # the name of the data-volume used by the container
+openldap_cv_name: '{{ docker_name_prefix }}ldap_CV'
 openldap_dv_name: '{{ docker_name_prefix }}ldap_DV'
 
 # restore directories to temporarly store data being restored into docker containers
diff --git a/roles/docker-openssl/tasks/main.yml b/roles/docker-openssl/tasks/main.yml
index 7c1b960..7d4ff5a 100644
--- a/roles/docker-openssl/tasks/main.yml
+++ b/roles/docker-openssl/tasks/main.yml
@@ -69,11 +69,9 @@
 # *****************************************************************************
 # Create the data volumes
 
-- name: data-volume container
-  docker_datavolume:
-    image_name: '{{ openssl_image_name }}'
-    image_tag: '{{ docker_image_tag }}'
-    data_volume_container_name: '{{ openssl_dv_name }}'
+- name: certificates volume
+  docker_volume:
+    name: '{{ openssl_dv_name }}'
 
 # *****************************************************************************
 # Populate the openssl data-volume with default data
diff --git a/roles/docker-openssl/templates/before_backup.j2 b/roles/docker-openssl/templates/before_backup.j2
index 3120162..d07e5ef 100644
--- a/roles/docker-openssl/templates/before_backup.j2
+++ b/roles/docker-openssl/templates/before_backup.j2
@@ -5,7 +5,8 @@
 docker run --rm -v {{ docker_backup_dir }}:/tmp/import_export:z debian:latest bash -c "rm -rf /tmp/import_export/openssl"
 mkdir -p -m u=rwx,g=rwx,o= {{ openssl_docker_backup_dir }}
 docker run --rm \
-    --volumes-from {{ openssl_dv_name }} \
+    --volume {{ openssl_dv_name }}:/etc/ssl/private:z \
     {{ openssl_image_name }}:{{ docker_image_tag }} backup > \
         {{ openssl_docker_backup_dir }}{{ openssl_backup_file }}
+
 {% endif %}
diff --git a/roles/docker-svn/tasks/main.yml b/roles/docker-svn/tasks/main.yml
index 00e687a..c7ae044 100644
--- a/roles/docker-svn/tasks/main.yml
+++ b/roles/docker-svn/tasks/main.yml
@@ -69,10 +69,8 @@
 # Create the data volumes
 
 - name: data-volume container
-  docker_datavolume:
-    image_name: '{{ svn_image_name }}'
-    image_tag: '{{ docker_image_tag }}'
-    data_volume_container_name: '{{ svn_dv_name }}'
+  docker_volume:
+    name: '{{ svn_dv_name }}'
 
 # *****************************************************************************
 # Start the data container running
@@ -86,9 +84,9 @@
     ports: '{{ container_port_map.svn.port_args }}'
     link:
       - '{{ openldap_container_name }}:ldap'
-    volumes-from:
-      - '{{ openssl_dv_name }}'
-      - '{{ svn_dv_name }}'
+    volumes:
+      - '{{ openssl_dv_name }}:/etc/ssl/private:z'
+      - '{{ svn_dv_name }}:/var/lib/svn:z'
     env:
       SVN_HOSTNAME: '{{ container_addr_map.svn.hostname }}'
       SVN_LDAP_PASSWORD: '{{ ldap_proxyagent_password }}'
diff --git a/roles/docker-svn/tasks/restore.yml b/roles/docker-svn/tasks/restore.yml
index 3e7b60e..34cee1e 100644
--- a/roles/docker-svn/tasks/restore.yml
+++ b/roles/docker-svn/tasks/restore.yml
@@ -38,9 +38,8 @@
 - name: restore svn repositories
   docker_container:
     detach: False
-    volumes-from:
-      - '{{ svn_dv_name }}'
     volumes:
+      - '{{ svn_dv_name }}:/var/lib/svn:z'
       - '{{ host_svn_docker_restore_dir }}:/tmp/import_export:z'
     image: '{{ svn_image_name }}:{{ docker_image_tag }}'
     command: ['import', '{{ svn_repos }}']
